experimenter/experimenter/nimbus_ui_new/templatetags/nimbus_extras.py

```diff
@@ -65,3 +67,21 @@ def format_not_set(value):
     if value in [None, "", "NOT SET"]:
         return mark_safe('<span class="text-danger">Not set</span>')
     return value
+
+
+@register.filter(name="format_json")
+def format_json(value):
+    """Formats the JSON value for display by pretty-printing the JSON,
+    and if the JSON is empty or None, it calls format_not_set."""
+    if value in [None, "", "NOT SET"]:
+        return format_not_set(value)
+
+    try:
+        parsed_json = json.dumps(json.loads(value), indent=2)
+    except (json.JSONDecodeError, TypeError):
+        parsed_json = value
+
+    return mark_safe(
+        '<pre class="text-monospace" style="white-space: pre-wrap; '
+        'word-wrap: break-word;">' + parsed_json + "</pre>"
```

Comment:
Oh right here, don't use concatenation to join strings 🙏 , use [fstrings](https://fstring.help/).